src/components/inline-inputs/inline-inputs.js

```diff
@@ -1,35 +1,29 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import { assign } from 'lodash';
 
 // Carbon
 import { Row, Column } from './../row';
 
-let InlineInputs = (props) => {
+const Label = (props) => {
+  if (!props.label) { return null; }
   return (
-    <div className={ classNames("carbon-inline-inputs", props.className) }>
-      { label(props) }
-      <Row gutter="none" className="carbon-inline-inputs__inputs">
-        { columnWrapper(props.children) }
-      </Row>
-    </div>
+    <label htmlFor={ props.htmlFor } className='carbon-inline-inputs__label'>
+      { props.label }
+    </label>
   );
 };
 
-let label = (props) => {
-  if (props.label) {
-    return <label className="carbon-inline-inputs__label">{ props.label }</label>;
-  }
-};
-
-let columnWrapper = (children) => {
+const columnWrapper = (children) => {
   let inputs = children;
 
   if (!Array.isArray(inputs)) {
     inputs = [children];
   }
 
   return inputs.map((input, index) => {
+    /* eslint-disable react/no-array-index-key */
```

Comment:
It is but the only way i can see to fix it is to expect a unique identifier on each of the inputs which is a breaking change. We could maybe look to address all these for version 2.0?